app/uk/gov/hmrc/helptosave/models/HTSEvent.scala

```diff
@@ -18,85 +18,144 @@ package uk.gov.hmrc.helptosave.models
 
 import cats.instances.int._
 import cats.syntax.eq._
+import play.api.libs.json._
 import uk.gov.hmrc.helptosave.config.AppConfig
 import uk.gov.hmrc.helptosave.controllers.routes
+import uk.gov.hmrc.helptosave.models.AccountCreated.{AllDetails, ExistingDetails, ManuallyEnteredDetails}
 import uk.gov.hmrc.helptosave.util.NINO
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.audit.AuditExtensions._
-import uk.gov.hmrc.play.audit.model.DataEvent
+import uk.gov.hmrc.play.audit.model.ExtendedDataEvent
 
 trait HTSEvent {
-  val value: DataEvent
+  val value: ExtendedDataEvent
 }
 
 object HTSEvent {
   def apply(appName:         String,
             auditType:       String,
-            detail:          Map[String, String],
+            detail:          JsValue,
             transactionName: String,
-            path:            String)(implicit hc: HeaderCarrier): DataEvent =
-    DataEvent(appName, auditType = auditType, detail = detail, tags = hc.toAuditTags(transactionName, path))
+            path:            String)(implicit hc: HeaderCarrier): ExtendedDataEvent =
+    ExtendedDataEvent(appName, auditType = auditType, detail = detail, tags = hc.toAuditTags(transactionName, path))
 }
 
 case class EligibilityCheckEvent(nino:              NINO,
                                  eligibilityResult: EligibilityCheckResult,
                                  ucResponse:        Option[UCResponse],
                                  path:              String)(implicit hc: HeaderCarrier, appConfig: AppConfig) extends HTSEvent {
 
-  val value: DataEvent = {
-    val details = {
+  val value: ExtendedDataEvent = {
+    val details: JsValue = {
+
       val result =
         if (eligibilityResult.resultCode === 1) {
-          Map[String, String]("nino" → nino, "eligible" → "true")
+          EligibilityResult(nino, true)
         } else {
           val reason = "Response: " +
             s"resultCode=${eligibilityResult.resultCode}, reasonCode=${eligibilityResult.reasonCode}, " +
             s"meaning result='${eligibilityResult.result}', reason='${eligibilityResult.reason}'"
-
-          Map[String, String]("nino" → nino, "eligible" → "false", "reason" -> reason)
-
+          EligibilityResult(nino, false, Some(reason))
         }
-      result ++ ucData(ucResponse)
+
+      ucData(result, ucResponse)
```

Comment:
what might be neater here is to have an apply method on `EligibilityResult` which looks like this:
```scala
object EligibilityResult{
  def apply(nino: String,  eligibilityResult: EligibilityCheckResult, ucResponse: Option[UCResponse])
}
```
rather than half constructing the case class with default values

Reply:
ok I've made changes to that section, thanks